packages/wallet/ui/src/util/WalletBackendAdapter.js

```diff
@@ -130,15 +130,71 @@ export const makeWalletBridgeFromFollower = (
     ]),
   );
 
+  // We assume just one cosmos purse per brand.
+  const offers = {};
+  const brandToPurse = new Map();
+
+  const updatePurses = () => {
+    const purses = [];
+    for (const purse of brandToPurse.values()) {
+      if (purse.currentAmount && purse.brandPetname) {
+        console.log(purse.currentAmount);
+        purses.push(purse);
+      }
+    }
+    // console.log(purses);
+    notifierKits.purses.updater.updateState(harden(purses));
+  };
+
   const followLatest = async () => {
-    for await (const { value: state } of iterateLatest(follower)) {
+    for await (const { value } of iterateEach(follower)) {
+      console.log('have value', value);
+      /** @type {import('@agoric/smart-wallet/src/smartWallet').UpdateRecord} */
+      const updateRecord = value;
       if (firstCallback) {
         firstCallback();
+        Object.values(notifierKits).forEach(({ updater }) =>
+          updater.updateState([]),
+        );
         firstCallback = undefined;
       }
-      Object.entries(notifierKits).forEach(([stateName, { updater }]) => {
-        updater.updateState(state[stateName]);
-      });
+      switch (updateRecord.updated) {
+        case 'brand': {
+          const { descriptor } = updateRecord;
+          const purseObj = {
+            ...brandToPurse.get(descriptor.brand),
+            brand: descriptor.brand,
+            brandPetname: descriptor.petname,
+            issuerPetname: descriptor.petname,
+            displayInfo: descriptor.displayInfo,
+          };
+          brandToPurse.set(descriptor.brand, purseObj);
+          updatePurses();
+          break;
+        }
+        case 'balance': {
+          // FIXME: We assume just one purse per brand.
```

Comment:
more of a TODO since it's not a requirement yet https://github.com/Agoric/agoric-sdk/issues/6126